src/org/zaproxy/zap/extension/replacer/ReplacerParam.java

```diff
@@ -182,6 +186,40 @@ public void setRules(List<ReplacerParamRule> rules) {
         enabledTokens.trimToSize();
     }
 
+    public ReplacerParamRule getRule(String desc) {
+    	for (ReplacerParamRule rule : rules) {
+    		if (rule.getDescription().equals(desc)) {
+    			return rule;
+    		}
+    	}
+        return null;
+    }
+    
+    public boolean setEnabled(String desc, boolean enabled) {
+    	ReplacerParamRule rule = this.getRule(desc);
+        if (rule != null) {
+        	rule.setEnabled(enabled);
+            this.saveRules();
+        	return true;
+        }
+    	return false;
+    }
+    
+    public void addRule(ReplacerParamRule rule) {
+    	this.rules.add(rule);
+    	this.saveRules();
+    }
+
+    public boolean removeRule(String desc) {
+    	ReplacerParamRule rule = this.getRule(desc);
+        if (rule != null) {
+        	this.rules.remove(rule);

```

Comment:
`ReplacerParamRule` needs to override equals, otherwise it will remove the rules based just on the enabled state.